src/controllers/songs.js

```diff
@@ -0,0 +1,72 @@
+const { Song } = require('../sequelize');
+const { Artist } = require('../sequelize');
+const { Album } = require('../sequelize');
+
+exports.createSongByAlbumId = (req, res) => {
+  const { albumId } = req.params;
+
+  Album.findByPk(albumId).then(album => {
+    if (!album) {
+      res.status(404).json({ error: 'The album could not be found.' });
+    } else {
+      Song.create({
+        name: req.body.name,
+        albumId: album.id,
+        artistId: album.artistId,
+      }).then(songCreated => {
+        res.status(201).json(songCreated);
+        console.log(songCreated);
+      });
+    }
+  });
+};
+exports.getSongsByAlbumId = (req, res) => {
+  const { albumId } = req.params;
+  Album.findByPk(albumId).then(song => {
+    if (!song) {
+      res.status(404).json({ error: 'The album could not be found.' });
+    } else {
+      Song.findAll({ where: { albumId }, include: [{ model: Artist, as: 'artist' }, { model: Album, as: 'album' }] }).then((songs) => {
+        res.status(200).json(songs);
+      });
+    }
+  });
+};
+
+exports.getSongsByArtistId = (req, res) => {
+  const { artistId } = req.params;
+  Artist.findByPk(artistId).then(artist => {
+    if (!artist) {
+      res.status(404).json({ error: 'The artist could not be found.' });
+    } else {
+      Song.findAll({ where: { artistId }, include: [{ model: Artist, as: 'artist' }, { model: Album, as: 'album' }] }).then((songs) => {
+        res.status(200).json(songs);
+      });
+    }
+  });
+};
+
+exports.updateSongById = (req, res) => {
+  const { songId } = req.params;
+  Song.update(req.body, { where: { id: songId } }).then(([rowsUpdated]) => {
+    if (!rowsUpdated) {
+      res.status(404).json({ error: 'The song could not be found.' });
+    } else {
+      res.status(200).json(rowsUpdated);
+    }
+  });
+};
+
+exports.deleteSongById = (req, res) => {
+  const { songId } = req.params;
+
+  Song.findByPk(songId).then(song => {
```

Comment:
I believe destroy return an array with the number of deleted records on it. You could use that to tell if you actually deleted anything. If it return `[0]` then you could assume that there was no song with that Id in the first place.